src/App.tsx

```diff
@@ -15,36 +15,91 @@
   'Garlic',
 ];
 
+type SelectedFilter = 'alphabetically' | 'length' | 'reverse' | '';
+
 export const App: React.FC = () => {
+  const [goods, setGoods] = useState(goodsFromServer);
+  const [selecteFilter, setSelectedFilter] = useState<SelectedFilter>('');
+  const [isReversed, setIsReversed] = useState(false);
+
+  function handleSortGoodsAlphabetically() {
+    if (isReversed) {
+      setGoods([...goods].sort((a, b) => b.localeCompare(a)));
+      setSelectedFilter('alphabetically');
+
+      return;
+    }
+
+    setGoods([...goods].sort((a, b) => a.localeCompare(b)));
+    setSelectedFilter('alphabetically');
+  }
+
+  function handleSortGoodsByLength() {
+    if (isReversed) {
+      setGoods([...goods].sort((a, b) => b.length - a.length));
+      setSelectedFilter('length');
+
+      return;
+    }
+
+    setSelectedFilter('length');
+    setGoods([...goods].sort((a, b) => a.length - b.length));
+  }
+
+  function handleReverseGoods() {
+    setGoods([...goods].reverse());
+    setIsReversed(!isReversed);
+  }
+
+  function handleResetGoods() {
+    setGoods(goodsFromServer);
+    setIsReversed(false);
+    setSelectedFilter('');
+  }
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          onClick={handleSortGoodsAlphabetically}
+          className={`button is-info ${selecteFilter === 'alphabetically' ? ' ' : 'is-light'}`}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          onClick={handleSortGoodsByLength}
+          className={`button is-success ${selecteFilter === 'length' ? ' ' : 'is-light'}`}
```

Comment:
The typo `selecteFilter` is used here. It should be corrected to `selectedFilter` to ensure the button's class is set correctly based on the filter state.